fix: [PIPE-19624]: Add changes to Reference By section (#6996)
harness/developer-hub: docs/platform/git-experience/configure-git-experience-for-harness-entities.md

To switch the referring child entity from a feature branch to a default branch, manually remove the field `gitBranch` from the parent YAML.

## Handling references for remote entities

> All entities have **Referenced By** tab where we can see which other entities are using are it.

Reviewer: ```suggestion
All entities have a **Referenced By** tab where you can view all other entities that use it.
```

> As part of remote entities, we only calculate references for the entities stored in the default branch.

Reviewer: ```suggestion
As part of remote entities, Harness only calculates references for the entities stored in the default branch.
```

#### Let's take an example of a Connector

When you click on **Connectors** there is a **Referenced by** tab which shows the entities that are using that connector including remote entities as well. 

![](./static/referenced_by.png)

> For example, If you store your entities—such as Pipelines, Services, Environment, Input Set, or Templates—in the default branch (e.g., main or master), they will appear in the **Referenced By** section. If a pipeline is using a connector, the connector will list the pipeline as a reference if the pipeline is stored remotely in the default branch or INLINE. However, if the pipeline is saved in a non-default branch, no reference will be created in the Referenced By section.

Reviewer: ```suggestion
For example, if you store your entities—such as Pipelines, Services, Environment, Input Set, or Templates—in the default branch (for example, main or master), they will appear in the **Referenced By** section. If a pipeline is using a connector, the connector will list the pipeline as a reference if the pipeline is stored remotely in the default branch or INLINE. However, if the pipeline is saved in a non-default branch, no reference will be created in the Referenced By section.
```

> For example, if you store your entities—such as Pipelines, Services, Environment, Input Set, or Templates—in the default branch (for example, main or master), they will appear in the **Referenced By** section. If a pipeline is using a connector, the connector will list the pipeline as a reference if the pipeline is stored remotely in the default branch or INLINE. However, if the pipeline is saved in a non-default branch, no reference will be created in the Referenced By section.

Reviewer: ```suggestion
For example, if you store your entities—such as Pipelines, Services, Environment, Input Set, or Templates—in the default branch (for example, main or master), they will appear in the **Referenced By** section. If a pipeline is using a connector, the connector will list the pipeline as a reference if the pipeline is stored remotely in the default branch or INLINE. However, if the pipeline is saved in a non-default branch, no reference will be created in the **Referenced By** section.
```
